apps/store/src/components/CartInventory/CartEntryItem.tsx

```diff
@@ -1,58 +1,124 @@
+          </DesktopPillowWithTitle>
         </div>
-        <SpaceFlex space={0.25}>
-          <DetailsSheetDialog {...cartEntry}>
-            <Dialog.Trigger asChild>
-              <Button variant="secondary" size="small">
-                {t('VIEW_ENTRY_DETAILS_BUTTON')}
-              </Button>
-            </Dialog.Trigger>
-          </DetailsSheetDialog>
-          <RemoveEntryDialog cartId={cartId} {...cartEntry}>
+        <Space y={1}>
+          <div>
+            <MobileTitle size="md">{title}</MobileTitle>
+            <Text size="md" color="textSecondary">
+              {/* @TODO: display "automatically switches" if cancellation is requested" */}
+              {startDate
+                ? t('CART_ENTRY_DATE_LABEL', { date: formatter.fromNow(startDate), ns: 'cart' })
+                : 'Starts sometime...'}
+            </Text>
+          </div>
+          <SpaceFlex space={0.25}>
+            <DetailsSheetDialog {...cartEntry}>
+              <Dialog.Trigger asChild>
+                <Button variant="secondary" size="small">
+                  {t('VIEW_ENTRY_DETAILS_BUTTON')}
+                </Button>
+              </Dialog.Trigger>
+            </DetailsSheetDialog>
             <Dialog.Trigger asChild>
-              <Button variant="ghost" size="small">
+              <Button variant="ghost" size="small" disabled={loading}>
                 {t('REMOVE_ENTRY_BUTTON')}
               </Button>
             </Dialog.Trigger>
-          </RemoveEntryDialog>
-        </SpaceFlex>
-      </Space>
-      <Text size="md">{formatter.monthlyPrice(cost)}</Text>
-    </Wrapper>
+          </SpaceFlex>
+        </Space>
+        <Text align="right" size="md">
+          {formatter.monthlyPrice(cost)}
+        </Text>
+      </Wrapper>
+
+      <FullscreenDialog.Modal
+        center
+        Footer={
+          <>
+            <Button
+              form={REMOVE_CART_ENTRY_FORM}
+              type="submit"
+              loading={loading}
+              disabled={loading}
+            >
+              {t('REMOVE_ENTRY_MODAL_CONFIRM_BUTTON')}
+            </Button>
+            <FullscreenDialog.Close asChild>
+              <Button type="button" variant="ghost">
+                {t('REMOVE_ENTRY_MODAL_CANCEL_BUTTON')}
+              </Button>
+            </FullscreenDialog.Close>
+          </>
+        }
+      >
+        <form id={REMOVE_CART_ENTRY_FORM} onSubmit={handleSubmit} />
+        <Text size={{ _: 'md', lg: 'xl' }} align="center">
+          {t('REMOVE_ENTRY_MODAL_PROMPT', { name: title })}
+        </Text>
+      </FullscreenDialog.Modal>
+    </FullscreenDialog.Root>
   )
 }
 
 const Wrapper = styled.li(({ theme }) => ({
   display: 'grid',
   gridTemplateColumns: '3rem minmax(0, 1fr) auto',
-  gap: theme.space[3],
+  gap: theme.space.sm,
+  [mq.lg]: {
+    gridTemplateColumns: '3fr 2fr 2fr',
+  },
+}))
+
+const DesktopTitle = styled(Text)({
+  display: 'none',
+  [mq.lg]: {
+    display: 'block',
```

Comment:
```suggestion
    display: 'revert',
```
This is just a small suggestion. I've seen this "pattern" a lot recently. The advantage is that one doesn't need to worry about checking what would be the default display value of an element.